lib/customer-deposit-wallet/src/Cardano/Wallet/Deposit/Pure/State/Creation.hs

```diff
@@ -1,45 +1,138 @@
     )
 
+import Cardano.Address.Derivation
+    ( xpubFromBytes
+    , xpubToBytes
+    )
 import Cardano.Crypto.Wallet
-    ( XPub
+    ( XPrv
+    , XPub
+    , generate
+    , toXPub
+    , unXPrv
     )
 import Cardano.Wallet.Deposit.Pure.State.Type
     ( WalletState (..)
     )
+import Data.Text
+    ( Text
+    )
 import Data.Word.Odd
     ( Word31
     )
+import GHC.Generics
+    ( Generic
+    )
 
 import qualified Cardano.Wallet.Deposit.Pure.Address as Address
 import qualified Cardano.Wallet.Deposit.Pure.Submissions as Sbm
 import qualified Cardano.Wallet.Deposit.Pure.UTxO.UTxOHistory as UTxOHistory
 import qualified Cardano.Wallet.Deposit.Read as Read
+import qualified Data.ByteString.Base16 as B16
+import qualified Data.ByteString.Char8 as B8
+import qualified Data.Text.Encoding as T
 
 data WalletPublicIdentity = WalletPublicIdentity
     { pubXpub :: XPub
     , pubNextUser :: Word31
     }
     deriving (Show)
 
-fromXPubAndGenesis
-    :: XPub -> Word31 -> Read.GenesisData -> WalletState
-fromXPubAndGenesis xpub knownCustomerCount genesisData =
+data Credentials
+    = XPubCredentials !XPub
+    | XPrvCredentials !XPrv !XPub
+    deriving (Generic, Show, Eq)
+
+instance Show XPrv where
+    show = B8.unpack . B16.encode . unXPrv
+
+instance Eq XPrv where
+    a == b = unXPrv a == unXPrv b
+
+xpubFromCredentials :: Credentials -> XPub
+xpubFromCredentials (XPubCredentials xpub) = xpub
+xpubFromCredentials (XPrvCredentials _ xpub) = xpub
+
+xprvFromCredentials :: Credentials -> Maybe XPrv
+xprvFromCredentials (XPubCredentials _) = Nothing
+xprvFromCredentials (XPrvCredentials xprv _) = Just xprv
+
+fromCredentialsAndGenesis
+    :: Credentials -> Word31 -> Read.GenesisData -> WalletState
+fromCredentialsAndGenesis credentials knownCustomerCount genesisData =
     WalletState
         { walletTip = Read.GenesisPoint
         , addresses =
-            Address.fromXPubAndCount network xpub knownCustomerCount
+            Address.fromXPubAndCount
+                network
+                (xpubFromCredentials credentials)
+                knownCustomerCount
         , utxoHistory = UTxOHistory.fromOrigin initialUTxO
         , txHistory = mempty
         , submissions = Sbm.empty
-        , rootXSignKey = Nothing
+        , rootXSignKey = xprvFromCredentials credentials
         }
   where
     network = Read.getNetworkId genesisData
     initialUTxO = mempty
+
+-- | Create 'Credentials' from a mnemonic sentence and a passphrase.
+credentialsFromMnemonics
+    :: Text
+    -- ^ Mnemonics
+    -> Text
+    -- ^ Passphrase
+    -> Credentials
+credentialsFromMnemonics mnemonics passphrase =
+    let
+        unencryptedXPrv =
+            generate
+                (T.encodeUtf8 mnemonics)
+                (T.encodeUtf8 mempty)
+        encryptedXPrv =
+            generate
+                (T.encodeUtf8 mnemonics)
+                (T.encodeUtf8 passphrase)
+    in
+        XPrvCredentials encryptedXPrv (toXPub unencryptedXPrv)
```

Comment:
Could you add a test that a signature with the `encryptedXPrv` and the passphrase will validate with the `toXPub unencryptedXPrv`? This is the thing that we were unsure about.